internal/experiments/experiments.go

```diff
@@ -0,0 +1,67 @@
+package experiments
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/joho/godotenv"
+)
+
+const envPrefix = "TASK_X_"
+
+var Flags struct{}
+
+func Parse() error {
+	if err := readDotEnv(); err != nil {
+		return err
+	}
+	t := reflect.TypeOf(&Flags)
+	v := reflect.ValueOf(&Flags)
```

Comment:
I'm confusing on what exactly you're trying to accomplish here. Can you elaborate?

Why do we need reflection? Couldn't this be a map/slice or similar just to store the name of the experiments that are enabled?

Reply:
No problem, the idea behind this was that we have a single struct definition for all our experiments. e.g.

```go
type Flags struct {
    GentleForce bool `x:"gentle force"`
}
```

I am then using reflection to look at each struct field and its tags to fetch the environment variables. Doing it in this way means that we don't need to add any custom code when experiments are added. They just need to be defined in the struct, in the same way you would for a struct that marshals JSON and are then accessible from anywhere.

I've pushed a commit that changes it to use a map instead (I'll squash this later). This means that users would need to specify the experiment via a string const rather than it being a struct. It's probably a little faster, but to check if an experiment is enabled you have to do:

```go
if experiments.IsEnabled(experiments.TestExperiment) {
	// do experiment stuff
}
```

Instead of 

```go
if experiments.Flags.TestExperiment {
        // do experiment stuff
}

Happy to move forwards with this if you prefer this approach.

Reply:
I now got what you tried to do.

In theory we could also do:

```go
var TestExperiment bool

func init() {
	TestExperiment = IsEnabled("TestExperiment")
}
```

or

```go
func IsEnabledTestExperiment() bool {
	return IsEnabled("TestExperiment")
}
```

I don't have any strong opinion between these three methods, but I think they are better than the first approach of using reflection. Any one of these three styles is fine to me.

Reply:
Actually, I really like the `init()` way. This gives the nicest possible API imo:

```go
if experiments.GentleForce {
    // do stuff
}
```

You also only need to update the file in 2 places this way when adding an experiment. I've updated this PR and the gentle force experiment PR.